autofill/autofill-impl/src/main/java/com/duckduckgo/autofill/impl/ui/credential/management/AutofillManagementRecyclerAdapterLegacy.kt

```diff
@@ -0,0 +1,271 @@
+import android.view.ViewGroup
+import androidx.lifecycle.LifecycleOwner
+import androidx.lifecycle.lifecycleScope
+import androidx.recyclerview.widget.RecyclerView
+import androidx.recyclerview.widget.RecyclerView.Adapter
+import com.duckduckgo.app.browser.favicon.FaviconManager
+import com.duckduckgo.autofill.api.domain.app.LoginCredentials
+import com.duckduckgo.autofill.impl.R
+import com.duckduckgo.autofill.impl.databinding.ItemRowAutofillCredentialsManagementScreenBinding
+import com.duckduckgo.autofill.impl.databinding.ItemRowAutofillCredentialsManagementScreenDividerBinding
+import com.duckduckgo.autofill.impl.databinding.ItemRowAutofillCredentialsManagementScreenHeaderBinding
+import com.duckduckgo.autofill.impl.databinding.ItemRowAutofillReportBreakageManagementScreenBinding
+import com.duckduckgo.autofill.impl.databinding.ItemRowSearchNoResultsBinding
+import com.duckduckgo.autofill.impl.ui.credential.management.AutofillManagementRecyclerAdapterLegacy.ContextMenuAction.CopyPassword
+import com.duckduckgo.autofill.impl.ui.credential.management.AutofillManagementRecyclerAdapterLegacy.ContextMenuAction.CopyUsername
+import com.duckduckgo.autofill.impl.ui.credential.management.AutofillManagementRecyclerAdapterLegacy.ContextMenuAction.Delete
+import com.duckduckgo.autofill.impl.ui.credential.management.AutofillManagementRecyclerAdapterLegacy.ContextMenuAction.Edit
+import com.duckduckgo.autofill.impl.ui.credential.management.sorting.CredentialGrouper
+import com.duckduckgo.autofill.impl.ui.credential.management.sorting.InitialExtractor
+import com.duckduckgo.autofill.impl.ui.credential.management.suggestion.SuggestionListBuilder
+import com.duckduckgo.autofill.impl.ui.credential.management.viewing.extractTitle
+import com.duckduckgo.autofill.impl.ui.credential.management.viewing.list.ListItem
+import com.duckduckgo.autofill.impl.ui.credential.management.viewing.list.ListItem.CredentialListItem.Credential
+import com.duckduckgo.autofill.impl.ui.credential.management.viewing.list.ListItem.CredentialListItem.SuggestedCredential
+import com.duckduckgo.autofill.impl.ui.credential.management.viewing.list.ListItem.Divider
+import com.duckduckgo.autofill.impl.ui.credential.management.viewing.list.ListItem.EmptyStateView
+import com.duckduckgo.autofill.impl.ui.credential.management.viewing.list.ListItem.GroupHeading
+import com.duckduckgo.autofill.impl.ui.credential.management.viewing.list.ListItem.NoMatchingSearchResults
+import com.duckduckgo.autofill.impl.ui.credential.management.viewing.list.ListItem.PromotionContainer
+import com.duckduckgo.autofill.impl.ui.credential.management.viewing.list.ListItem.ReportAutofillBreakage
+import com.duckduckgo.autofill.impl.ui.credential.management.viewing.list.ListItem.TopLevelControls
+import com.duckduckgo.common.ui.menu.PopupMenu
+import com.duckduckgo.common.utils.DispatcherProvider
+import kotlinx.coroutines.launch
+import kotlinx.coroutines.withContext
+
+class AutofillManagementRecyclerAdapterLegacy(
```

Comment:
This class is as touched as possible and is largely just what is `AutofillManagementRecyclerAdapter` today